Formula/rustfmt.rb

```diff
@@ -25,8 +25,8 @@ def install
     system "#{Formula["rustup-init"].bin}/rustup-init", "-qy", "--no-modify-path"
     ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
     # we are using nightly because rustfmt requires nightly in order to build from source
-    # pinning to nightly-2021-11-08 to avoid inconstency
-    nightly_version = "nightly-2021-11-08"
+    # pinning to nightly-2023-01-24 to avoid inconstency
+    nightly_version = "nightly-2023-01-24"
```

Comment:
I wonder if we can just read this from the `rust-toolchain` file so that this doesn't have to be updated manually.

Reply:
that makes sense.

Reply:
note rustup's `cargo` already [reads the version](https://rust-lang.github.io/rustup/overrides.html) from that file, so unless we want to be explicit all we need is

```ruby
    system "#{Formula["rustup-init"].bin}/rustup-init", "-qy", "--no-modify-path"
    ENV.prepend_path "PATH", HOMEBREW_CACHE/"cargo_cache/bin"
    system "cargo", "install", *std_cargo_args
```

<details>
<summary><b>Example</b></summary>
<p>

```console
==> cargo install --locked --root /opt/homebrew/Cellar/rustfmt/1.5.2 --path .
info: syncing channel updates for 'nightly-2023-01-24-aarch64-apple-darwin'
info: latest update on 2023-01-24, rust version 1.69.0-nightly (c8e6a9e8b 2023-01-23)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'llvm-tools'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
info: downloading component 'rustc-dev'
info: downloading component 'rustfmt'
info: installing component 'cargo'
info: installing component 'clippy'
info: installing component 'llvm-tools'
info: installing component 'rust-docs'
info: installing component 'rust-std'
info: installing component 'rustc'
info: installing component 'rustc-dev'
info: installing component 'rustfmt'
  Installing rustfmt-nightly v1.5.2 (/private/tmp/rustfmt-20230126-48518-ucx7qv/rustfmt-1.5.2)
...
```

</p>
</details>